core/src/main/java/com/google/zxing/oned/Code93Writer.java

```diff
@@ -40,74 +40,49 @@ public BitMatrix encode(String contents,
 
   @Override
   public boolean[] encode(String contents) {
+    contents = convertToExtended(contents);
     int length = contents.length();
     if (length > 80) {
       throw new IllegalArgumentException(
-        "Requested contents should be less than 80 digits long, but got " + length);
+        "Requested contents should be less than 80 digits long after extended, but got " + length);
     }
-    //each character is encoded by 9 of 0/1's
-    int[] widths = new int[9];
 
     //length of code + 2 start/stop characters + 2 checksums, each of 9 bits, plus a termination bar
     int codeWidth = (contents.length() + 2 + 2) * 9 + 1;
 
-    //start character (*)
-    toIntArray(Code93Reader.CHARACTER_ENCODINGS[47], widths);
-
     boolean[] result = new boolean[codeWidth];
-    int pos = appendPattern(result, 0, widths);
+
+    //start character (*)
+    int pos = appendPattern(result, 0, Code93Reader.ASTERISK_ENCODING);
 
     for (int i = 0; i < length; i++) {
       int indexInString = Code93Reader.ALPHABET_STRING.indexOf(contents.charAt(i));
-      toIntArray(Code93Reader.CHARACTER_ENCODINGS[indexInString], widths);
-      pos += appendPattern(result, pos, widths);
+      pos += appendPattern(result, pos, Code93Reader.CHARACTER_ENCODINGS[indexInString]);
     }
 
     //add two checksums
     int check1 = computeChecksumIndex(contents, 20);
-    toIntArray(Code93Reader.CHARACTER_ENCODINGS[check1], widths);
-    pos += appendPattern(result, pos, widths);
+    pos += appendPattern(result, pos, Code93Reader.CHARACTER_ENCODINGS[check1]);
 
     //append the contents to reflect the first checksum added
     contents += Code93Reader.ALPHABET_STRING.charAt(check1);
 
     int check2 = computeChecksumIndex(contents, 15);
-    toIntArray(Code93Reader.CHARACTER_ENCODINGS[check2], widths);
-    pos += appendPattern(result, pos, widths);
+    pos += appendPattern(result, pos, Code93Reader.CHARACTER_ENCODINGS[check2]);
 
     //end character (*)
-    toIntArray(Code93Reader.CHARACTER_ENCODINGS[47], widths);
-    pos += appendPattern(result, pos, widths);
+    pos += appendPattern(result, pos, Code93Reader.ASTERISK_ENCODING);
 
     //termination bar (single black bar)
     result[pos] = true;
 
     return result;
   }
 
-  private static void toIntArray(int a, int[] toReturn) {
+  private static int appendPattern(boolean[] target, int pos, int a) {
     for (int i = 0; i < 9; i++) {
       int temp = a & (1 << (8 - i));
-      toReturn[i] = temp == 0 ? 0 : 1;
-    }
-  }
-
-  /**
-   * @param target output to append to
-   * @param pos start position
-   * @param pattern pattern to append
-   * @param startColor unused
-   * @return 9
-   * @deprecated without replacement; intended as an internal-only method
-   */
-  @Deprecated
-  protected static int appendPattern(boolean[] target, int pos, int[] pattern, boolean startColor) {
```

Comment:
This shouldn't have been protected in the first place, or else yes I'd have removed it. I kept it to maintain full backwards-compatibility, even if I don't see any reason for a caller to use this. There is a check in the build that I thought would fail if a protected or public method was removed, but, this passed. I still think maybe it's best to keep the method, just to minimize change and to remove any possibility of a maintenance release breaking some code.